src/Parser/Shaman/Restoration/Modules/Spells/LavaSurge.js

```diff
@@ -0,0 +1,22 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+
+
+class LavaSurge extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  on_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.LAVA_SURGE.id) {
+      return;
+    }
+    if (this.spellUsable.isOnCooldown(SPELLS.LAVA_BURST.id)) {
+      this.spellUsable.reduceCooldown(SPELLS.LAVA_BURST.id, 8000);
```

Comment:
I'd suggest you define the cooldown duration as a global constant before the class:
`const LAVA_BURST_CD_MS = 8000;`
Same with riptide CD in EchoReset.

Reply:
Actually, you could also get the expected CD from abilities instead, that way you'd avoid the cooldown being hardcoded in two different places.